Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs

```diff
@@ -0,0 +1,317 @@
+
+      return Analyze ($"#{controlObject.GetHtmlID()}");
+    }
+
+    /// <summary>
+    /// Analyzes a specific element on webpage.
+    /// </summary>
+    /// <param name="cssSelector">CSS selector of the ControlObjects to analyze.</param>
+    /// <returns> Results of the analysis.</returns>
+    [NotNull]
+    public AccessibilityResult Analyze ([NotNull] string cssSelector)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("cssSelector", cssSelector);
+
+      return GetAccessibilityResult (cssSelector);
+    }
+
+    /// <summary>
+    /// Analyzes all content on a webpage.
+    /// Switches to the outermost frame before the analysis and back to
+    /// the original frame after the analysis completed.
+    /// </summary>
+    /// <returns> Results of the analysis.</returns>
+    [NotNull]
+    public AccessibilityResult Analyze ()
+    {
+      var outerFrame = (string) JsExecutor.ExecuteScript ("return self.name;");
+      if (outerFrame != "")
+        WebDriver.SwitchTo().DefaultContent();
+
+      var result = GetAccessibilityResult (null);
+      if (outerFrame != "")
+        WebDriver.SwitchTo().Frame (outerFrame);
+
+      return result;
+    }
+
+    private AccessibilityResult GetAccessibilityResult (string cssSelector)
+    {
+      var source = AxeSourceProvider.GetSource();
+
+      if (!AxeIsInjected())
+      {
+        using (new PerformanceTimer (Logger, "aXe has been injected."))
+        {
+          InjectAxeSource (source);
+        }
+      }
+
+      var axeRunFunctionCall = BuildAxeRunFunctionCall (cssSelector);
+
+      string result;
+      using (new PerformanceTimer (Logger, "Accessibility analysis has been performed."))
+      { 
+        result = (string) JsExecutor.ExecuteAsyncScript (axeRunFunctionCall) ?? GetRawAxeResultForInternetExplorer (axeRunFunctionCall);
+      }
+
+      var parsedResult = AxeResultParser.Parse (result);
+
+      return Mapper.Map (parsedResult);
+    }
+
+    private bool AxeIsInjected ()
+    {
+      return (bool) JsExecutor.ExecuteScript ("return (typeof axe !== 'undefined')");
+    }
+
+    // Injects Axe into every iframe manually because axe-core cannot do it itself
+    // https://www.deque.com/blog/writing-automated-tests-accessibility/
+    private void InjectIntoIFrames (string source)
+    {
+      foreach (var frame in WebDriver.FindElements (By.TagName ("iframe")))
+      {
+        WebDriver.SwitchTo().Frame (frame);
+
+        if (!IsEmptyFrame() && !AxeIsInjected())
+          InjectAxeSource (source);
+
+        WebDriver.SwitchTo().DefaultContent();
+      }
+    }
+
+    private bool IsEmptyFrame ()
+    {
+      return !WebDriver.FindElements (By.XPath ("/html/body/*")).Any();
+    }
+
+    private void InjectAxeSource (string source)
+    {
+      JsExecutor.ExecuteScript (source);
+
+      if (Configuration.IncludeIFrames)
+        InjectIntoIFrames (source);
+    }
+
+    private string BuildAxeRunFunctionCall (string cssToInclude)
```

Comment:
should we name the methods receiving a stringbuilder 'append' instead of 'build'?